tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td

```diff
@@ -684,3 +684,20 @@ def OptimizeMul1ToIdentity : Pat<
   (replaceWithValue $input),
   [(HaveSameType $input, $constant),
    (AllElementsAre<"1.0f"> $constant)]>;
+
+// Remove (log-)softmax before arg-minmax as (log-)softmax is monotonic.
```

Comment:
Can we move these op lowering under lower_tf.cc or lower_tf.td in the tensorflow mlir directory?

Reply:
There is an optimization pattern for TensorFlow MLIR dialect:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/tensorflow/transforms/optimize.td

Reply:
Thanks for pointing out! My concern is that there are some patterns that converts TFL ops into TFL_Softmax. See `OptimizeToSoftmax` and `FoldNormalizationIntoSoftmax`. If we put this pattern into TF dialect, it won't take effect for those optimizations cause the ops have been legalized. Is there any workaround?

Reply:
Do you think `OptimizeToSoftmax` also can be moved to the optimize.td under the TensorFlow MLIR without causing any impacts on the existing models?

Reply:
If not, I think the current approach will be fine.

Reply:
I'm afraid it's hard to do that without breaking existing models because there are more ops used in `OptimizeToSoftmax`, such as TFL_Sum and TFL_Exp, and those ops are involved in other patterns. That is, we might have to move more patterns than just one. I'd say that most optimization patterns in TFL are suitable in TF as well (regardless XLA); probably a long-term plan is to migrate those compatible patterns into TF. Thank you!